tests/res_diff_expect/fail_read_compare_vs_other_errors_2.ion

```diff
@@ -0,0 +1,46 @@
+$ion_1_0
+{
+ test: {
+  read_compare: {
+   message: "Read_compare: two revisions have different failures.",
+   failures: {
+    'ion-java_1': {
+     failures: ComparisonReport::[
+      {
+       result: "NOT_EQUAL",
+       message: "two event streams have different size",
+       lhs: {
+        location: "ion-test-driver/amazon/iontest/results/good/test.ion/read/data/ion-java_1.ion"
+       },
+       rhs: {
+        location: "ion-test-driver/amazon/iontest/results/good/test.ion/read/data/ion-java_2.ion"
+       }
+      },
+      {
+       result: "NOT_EQUAL",
+       message: "two event streams have different size",
+       lhs: {
+        location: "ion-test-driver/amazon/iontest/results/good/test.ion/read/data/ion-java_1.ion"
+       },
+       rhs: {
+        location: "ion-test-driver/amazon/iontest/results/good/test.ion/read/data/ion-c_1.ion"
+       }
+      },
+      {
+       result: "NOT_EQUAL",
+       message: "two event streams have different size",
+       lhs: {
+        location: "ion-test-driver/amazon/iontest/results/good/test.ion/read/data/ion-java_2.ion"
+       },
+       rhs: {
+        location: "ion-test-driver/amazon/iontest/results/good/test.ion/read/data/ion-js_1.ion"
+       }
+      }
+     ]
+    },
+    'ion-java_2': {
+    }
```

Comment:
We check report phase by phase and end analyzing the file once we found errors.
For this test case that ion-java_2 has a phase three error but the analysis detects errors in phase two (one has read_compare error and the other one not), so analysis for this file end in phase two and we don't have phase three errors included in ion-java_2 report.

The reason to end it immediately once we find an error is that because if one phase has already had an error, the later phase error may not be accurate or caused by earlier phase so we don't spend unnecessary time on analysis. Do you think it's a good strategy?

Reply:
Okay, let's stick with that for now then.